src/firefly/js/charts/ui/ChartSelectPanel.jsx

```diff
@@ -106,60 +106,46 @@ function getGroupKey(chartId, chartAction) {
     }
 }
 
-
-export class ChartSelectPanel extends SimpleComponent {
-    constructor(props) {
-        super(props);
-
-        const {chartId, tbl_id, chartAction} = this.props;
-        let newChartAction = chartAction;
-        const chartActions = getChartActions({chartId, tbl_id});
-
-        if (!newChartAction || !chartActions.includes(newChartAction)) {
-            newChartAction = chartActions[0];
-        }
-
-        this.state = {chartAction: newChartAction};
-    }
-
-    render() {
-        const {tbl_id, chartId, inputStyle={}, hideDialog, style={}} = this.props;
-        const {renderTreeId}= this.context;
-
-        const chartActions = getChartActions({chartId, tbl_id});
-        const {chartAction} = this.state;
-        const groupKey = getGroupKey(chartId, chartAction);
-
-        const chartActionChanged = (chartAction) => { this.setState({chartAction}); };
-
-        return (
-            <div style={{padding: 10, ...style}}>
-                <FormPanel
-                    groupKey={groupKey}
-                    submitText={chartAction===CHART_TRACE_MODIFY ? 'Apply' : 'OK'}
-                    onSuccess={onChartAction({chartAction, tbl_id, chartId, hideDialog, renderTreeId})}
-                    cancelText='Close'
-                    onError={() => {}}
-                    onCancel={hideDialog}
-                    inputStyle = {inputStyle}
-                    style={{padding:5}}
-                    changeMasking={this.changeMasking}>
-                    <ChartAction {...{chartId, chartActions, chartAction, chartActionChanged}}/>
-                    <ChartActionOptions {...{chartAction, tbl_id, chartId, groupKey, hideDialog}}/>
-                </FormPanel>
-            </div>
-        );
-    }
+export function ChartSelectPanel({tbl_id, chartId, chartAction, inputStyle={}, hideDialog, style={}}) {
+    const {renderTreeId} = useContext(RenderTreeIdCtx);
+
+    const chartActions = getChartActions({chartId, tbl_id});
+    const [chartActionState, setChartActionState] = useState(
+        (!chartAction || !chartActions.includes(chartAction)) ? chartActions[0] : chartAction);
+
+    const groupKey = getGroupKey(chartId, chartActionState);
+
+    const chartActionChanged = (chartAction) => setChartActionState(chartAction);
+
+    return (
+        <div style={{padding: 10, ...style}}>
+            <FormPanel
+                groupKey={groupKey}
+                submitText={chartActionState===CHART_TRACE_MODIFY ? 'Apply' : 'OK'}
+                onSuccess={onChartAction({chartAction: chartActionState,
+                    tbl_id, chartId, hideDialog, renderTreeId})}
+                cancelText='Close'
+                onError={() => {}}
```

Comment:
I don't think this is correct. It returns an undefined not an empty object. Maybe you want to return undefined. Either way it should be clear.

for object
```js 
onError={() => ({})}
```
for undefined
```js
onError={() => undefined}
```

Reply:
Well it's actually a void function that does nothing. The FormPanel component also supplies it as default for onError so I think we can omit it from here altogether.